src/adapter.rs

```diff
@@ -471,6 +473,118 @@ impl Adapter {
+            assert_ne!(res, 0); // 0 means we got the size wrong
+            let wireguard_interface: WIREGUARD_INTERFACE = reader.read();
+            let mut wg_interface = WireguardInterface {
+                Flags: wireguard_interface.Flags as u32,
+                ListenPort: wireguard_interface.ListenPort,
+                PrivateKey: wireguard_interface.PrivateKey,
+                PublicKey: wireguard_interface.PublicKey,
+                Peers: Vec::with_capacity(wireguard_interface.PeersCount as usize),
+            };
+            for _ in 1..=wireguard_interface.PeersCount {
+                let peer: WIREGUARD_PEER = reader.read();
+                let endpoint = peer.Endpoint;
+                let endpoint = match endpoint.si_family as i32 {
+                    winapi::shared::ws2def::AF_INET => {
+                        let octets = endpoint.Ipv4.sin_addr.S_un.S_un_b;
+                        let address = Ipv4Addr::new(octets.s_b1, octets.s_b2, octets.s_b3, octets.s_b4);
+                        let port = u16::from_be(endpoint.Ipv4.sin_port);
+                        SocketAddr::V4(SocketAddrV4::new(address, port))
+                    }
+                    winapi::shared::ws2def::AF_INET6 => {
+                        let octets = endpoint.Ipv6.sin6_addr.u.Byte;
+                        let address = Ipv6Addr::from(octets);
+                        let port = u16::from_be(endpoint.Ipv6.sin6_port);
+                        let flow_info = endpoint.Ipv6.sin6_flowinfo;
+                        let scope_id = endpoint.Ipv6.__bindgen_anon_1.sin6_scope_id;
+                        SocketAddr::V6(SocketAddrV6::new(address, port, flow_info, scope_id))
+                    }
+                    _ => { panic!("Illegal address family {}", endpoint.si_family); }
+                };
+                let mut wg_peer = WireguardPeer {
+                    Flags: peer.Flags as u32,
+                    PublicKey: peer.PublicKey,
+                    PresharedKey: peer.PresharedKey,
+                    PersistentKeepalive: peer.PersistentKeepalive,
+                    Endpoint: endpoint,
+                    TxBytes: peer.TxBytes,
+                    RxBytes: peer.RxBytes,
+                    LastHandshake: peer.LastHandshake,
+                    AllowedIps: Vec::with_capacity(peer.AllowedIPsCount as usize),
+                };
+                for _ in 1..=peer.AllowedIPsCount {
+                    let allowed_ip: WIREGUARD_ALLOWED_IP = reader.read();
+                    let prefix_length = allowed_ip.Cidr;
+                    let allowed_ip = match allowed_ip.AddressFamily as i32 {
+                        winapi::shared::ws2def::AF_INET => {
+                            let octets = allowed_ip.Address.V4.S_un.S_un_b;
+                            let address = Ipv4Addr::new(octets.s_b1, octets.s_b2, octets.s_b3, octets.s_b4);
```

Comment:
A transmute might be nicer here. Both WIN32 and Rust use big endian so A.B.C.D is stored as 0x0A0B0C0D. What do you think?

Reply:
I agree it would be a cool trick here, but I guess if we can avoid another unsafe method call, maybe that's the right thing to do.

Reply:
Actually you're right. This should work for both little and big endian.